green-square-sequence/src/App.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react'
+import 'style.css'
+
+//set how many green squares you want displayed here
+let numOfSquares = 6
+
+export default function App() {
+  let [order, setOrder] = useState([])
+  let squares = [...Array(numOfSquares)]
+
+  const addElement = (num) => {
+    setOrder((order = order.concat(num)))
+  }
+
+  const reverseSequence = async () => {
+    for (let i = 1; i <= numOfSquares; i++) {
+      await delay(500)
+      const newOrder = order.slice(0, order.length - i)
+      setOrder(newOrder)
+    }
```

Comment:
This loop is good --  but there's a tricky edge case to consider.

Imagine this component was a little minigame embedded in a larger app. A user gets to the game, clicks all squares, the reversing animation starts playing, then the user closes the component before the animation finishes. The loop would still be running in the background firing setOrder calls to an unmounted component.

It doesn't _really_ matter in this example, because the timer is finite and short. But if you had a component that was constantly animating this pattern would be much worse.

This is a bit icky to do properly in React -- useEffect and/or useRef will likely help.